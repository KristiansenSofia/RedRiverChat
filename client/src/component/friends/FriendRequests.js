import React, { Component } from 'react';
import { Redirect } from 'react-router-dom';

// Import styles. ChatListStyles for all imported components with a style attribute and CSS-file for classNames and id.
import {friendRequestStyles} from "../../styles/FriendsStyles";
import '../../styles/Styles.css'

/**
 *  FriendsList-component. Starting page of friends.
 *
 *  @author Jimmy
 */

class FriendRequests extends Component {
    render() {

        if (this.props.state.isSignedIn === false) {
            return <Redirect to="/login" />
        }

        return (
            <div className="FriendsRequests">
                <p>Friend Requests</p>
            </div>

        );
    }
}

export default FriendRequests;