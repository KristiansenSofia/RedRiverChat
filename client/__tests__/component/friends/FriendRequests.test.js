import React from 'react';
import Enzyme from 'enzyme';
import {mount, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import FriendRequests from '../../../src/component/friends/FriendRequests';
import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';

Enzyme.configure({ adapter: new Adapter() });

describe('FriendRequests', () => {
  const mockToken = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiJCYW5hbiIsImVtYWlsIjoiYmFuYW5AZXhhbXBsZS5jb20iLCJqdGkiOiI3MWE1YzA3Yy1mNWNhLTQ3MTYtYjk0Ny05ZDlhZWIwNjZmNzciLCJyb2xlcyI6InVzZXIiLCJleHAiOjE1MjQ2NTE5MzIsImlzcyI6Imh0dHA6Ly9sb2NhbGhvc3Q6NjM5MzkvIiwiYXVkIjoiaHR0cDovL2xvY2FsaG9zdDo2MzkzOS8ifQ.C_xboFny2nszame4GHJNnlOzlWJ2tVGmflfnmJAyXAE';

  const props = {
    openSnackBar: jest.fn(),
    state: {
      isSignedIn: true,
      token: mockToken
    }
  };

  const component = shallow(
    <FriendRequests {...props}/>
  );

  it('should render without crashing', () => {
    expect(component).toMatchSnapshot();
  });


  describe('handleChange', () => {
    it('should set friendUserName to Banan', () => {
      let event = {
        target: {
          value: 'Banan'
        }
      }

      component.instance().handleChange('friendUserName')(event);
      const friendUserNameState = component.state('friendUserName');

      expect(friendUserNameState).toEqual('Banan');
    });

    // TODO: This is not implemented in the source code yet.
    /*it('should set friendRequests to Jan and test1', () => {
      let friendsRequests = ['Jan', 'test1'];

      let event1 = {
        target: {
          value: 'Jan'
        }
      }

      let event2 = {
        target: {
          value: 'test1'
        }
      }

      component.instance().handleChange('friendRequests')(event1);
      component.instance().handleChange('friendRequests')(event2);
      const friendRequestsState = component.state('friendRequests');

      expect(friendRequestsState).toEqual(friendsRequests);
    });*/
  });


  describe('handleSubmit', () => {
    const sendRequestSpy = jest.spyOn(FriendRequests.prototype, 'sendRequest');

    it('should return "FormulÃ¤ret ej korrekt ifyllt!" when friendUserName is missing', (done) => {
      component.setState({
        friendUserName: ''
      });

      component.instance().handleSubmit();

      expect(props.openSnackBar).toHaveBeenCalledTimes(1);
      expect(sendRequestSpy).toHaveBeenCalledTimes(0);
      done();
    });


    it('should call sendRequest', (done) => {
      component.setState({
        friendUserName: 'Banan'
      });

      component.instance().handleSubmit();

      expect(sendRequestSpy).toHaveBeenCalledTimes(1);
      done();
    });
  });


  describe('sendRequest', () => {
    const apiUrl = 'https://redserver.azurewebsites.net/api/user/addfriend';

    it('should respond with status 200 when user exist', (done) => {
      let mock = new MockAdapter(axios);
      mock.onPost(apiUrl).reply(200);

      component.setState({
        friendUserName: 'test1'
      });

      component.instance().sendRequest().then(response => {
        expect(response.status).toEqual(200);
        done();
      });
    });


    it('should respond with status 404 when user does not exist', (done) => {
      let mock = new MockAdapter(axios);
      mock.onGet(apiUrl).reply(404);

      component.setState({
        friendUserName: 'BananApan'
      });

      component.instance().sendRequest().catch(error => {
        expect(error.message).toEqual('Request failed with status code 404');
        expect(error.response.status).toEqual(404);
        done();
      });
    });
  });


  describe('render', () => {
    it('should navigate to login when user is not signed in', () => {
      const component1 = shallow(
        <FriendRequests state = {{ isSignedIn: false }}/>
      );

      component1.instance().render();

      expect(component1).toMatchObject(/Redirect to="\/login"/);
    });
  });
});
