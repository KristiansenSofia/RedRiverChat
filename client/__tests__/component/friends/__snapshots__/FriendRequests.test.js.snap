// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FriendRequests should render without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FriendRequests
    openSnackBar={[MockFunction]}
    state={
      Object {
        "isSignedIn": true,
        "token": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiJCYW5hbiIsImVtYWlsIjoiYmFuYW5AZXhhbXBsZS5jb20iLCJqdGkiOiI3MWE1YzA3Yy1mNWNhLTQ3MTYtYjk0Ny05ZDlhZWIwNjZmNzciLCJyb2xlcyI6InVzZXIiLCJleHAiOjE1MjQ2NTE5MzIsImlzcyI6Imh0dHA6Ly9sb2NhbGhvc3Q6NjM5MzkvIiwiYXVkIjoiaHR0cDovL2xvY2FsaG9zdDo2MzkzOS8ifQ.C_xboFny2nszame4GHJNnlOzlWJ2tVGmflfnmJAyXAE",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <WithStyles(Typography)
          align="left"
          color="default"
          style={
            Object {
              "margin": 20,
            }
          }
          variant="headline"
        >
          Vänförfrågningar
        </WithStyles(Typography)>,
        <div
          className="FriendRequests-Inner"
        >
          <WithStyles(Typography)>
            Inga nya vänförfrågningar
          </WithStyles(Typography)>
        </div>,
        <WithStyles(Divider) />,
        <div
          className="FriendRequests-Inner"
        >
          <TextField
            id="userName"
            label="Lägg till ny vän:"
            margin="normal"
            onChange={[Function]}
            required={false}
            select={false}
            style={
              Object {
                "margin": 10,
              }
            }
            value=""
          />
          <WithStyles(Button)
            onClick={[Function]}
            style={
              Object {
                "margin": 30,
              }
            }
            variant="raised"
          >
            Skicka vänförfrågan
          </WithStyles(Button)>
        </div>,
      ],
      "className": "FriendRequests",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "align": "left",
          "children": "Vänförfrågningar",
          "color": "default",
          "style": Object {
            "margin": 20,
          },
          "variant": "headline",
        },
        "ref": null,
        "rendered": "Vänförfrågningar",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <WithStyles(Typography)>
            Inga nya vänförfrågningar
          </WithStyles(Typography)>,
          "className": "FriendRequests-Inner",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Inga nya vänförfrågningar",
          },
          "ref": null,
          "rendered": "Inga nya vänförfrågningar",
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <TextField
              id="userName"
              label="Lägg till ny vän:"
              margin="normal"
              onChange={[Function]}
              required={false}
              select={false}
              style={
                Object {
                  "margin": 10,
                }
              }
              value=""
            />,
            <WithStyles(Button)
              onClick={[Function]}
              style={
                Object {
                  "margin": 30,
                }
              }
              variant="raised"
            >
              Skicka vänförfrågan
            </WithStyles(Button)>,
          ],
          "className": "FriendRequests-Inner",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "id": "userName",
              "label": "Lägg till ny vän:",
              "margin": "normal",
              "onChange": [Function],
              "required": false,
              "select": false,
              "style": Object {
                "margin": 10,
              },
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Skicka vänförfrågan",
              "onClick": [Function],
              "style": Object {
                "margin": 30,
              },
              "variant": "raised",
            },
            "ref": null,
            "rendered": "Skicka vänförfrågan",
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <WithStyles(Typography)
            align="left"
            color="default"
            style={
              Object {
                "margin": 20,
              }
            }
            variant="headline"
          >
            Vänförfrågningar
          </WithStyles(Typography)>,
          <div
            className="FriendRequests-Inner"
          >
            <WithStyles(Typography)>
              Inga nya vänförfrågningar
            </WithStyles(Typography)>
          </div>,
          <WithStyles(Divider) />,
          <div
            className="FriendRequests-Inner"
          >
            <TextField
              id="userName"
              label="Lägg till ny vän:"
              margin="normal"
              onChange={[Function]}
              required={false}
              select={false}
              style={
                Object {
                  "margin": 10,
                }
              }
              value=""
            />
            <WithStyles(Button)
              onClick={[Function]}
              style={
                Object {
                  "margin": 30,
                }
              }
              variant="raised"
            >
              Skicka vänförfrågan
            </WithStyles(Button)>
          </div>,
        ],
        "className": "FriendRequests",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "align": "left",
            "children": "Vänförfrågningar",
            "color": "default",
            "style": Object {
              "margin": 20,
            },
            "variant": "headline",
          },
          "ref": null,
          "rendered": "Vänförfrågningar",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <WithStyles(Typography)>
              Inga nya vänförfrågningar
            </WithStyles(Typography)>,
            "className": "FriendRequests-Inner",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Inga nya vänförfrågningar",
            },
            "ref": null,
            "rendered": "Inga nya vänförfrågningar",
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <TextField
                id="userName"
                label="Lägg till ny vän:"
                margin="normal"
                onChange={[Function]}
                required={false}
                select={false}
                style={
                  Object {
                    "margin": 10,
                  }
                }
                value=""
              />,
              <WithStyles(Button)
                onClick={[Function]}
                style={
                  Object {
                    "margin": 30,
                  }
                }
                variant="raised"
              >
                Skicka vänförfrågan
              </WithStyles(Button)>,
            ],
            "className": "FriendRequests-Inner",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "id": "userName",
                "label": "Lägg till ny vän:",
                "margin": "normal",
                "onChange": [Function],
                "required": false,
                "select": false,
                "style": Object {
                  "margin": 10,
                },
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Skicka vänförfrågan",
                "onClick": [Function],
                "style": Object {
                  "margin": 30,
                },
                "variant": "raised",
              },
              "ref": null,
              "rendered": "Skicka vänförfrågan",
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
